src/test/HostActivationTests/FrameworkResolution/RollForwardMultipleFrameworks.cs

```diff
@@ -0,0 +1,544 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.Cli.Build;
+using Microsoft.DotNet.Cli.Build.Framework;
+using System;
+using Xunit;
+
+namespace Microsoft.DotNet.CoreSetup.Test.HostActivation.FrameworkResolution
+{
+    public class RollForwardMultipleFrameworks :
+        FrameworkResolutionBase,
+        IClassFixture<RollForwardMultipleFrameworks.SharedTestState>
+    {
+        private const string MiddleWare = "MiddleWare";
+        private const string AnotherMiddleWare = "AnotherMiddleWare";
+        private const string HighWare = "HighWare";
+
+        private SharedTestState SharedState { get; }
+
+        public RollForwardMultipleFrameworks(SharedTestState sharedState)
+        {
+            SharedState = sharedState;
+        }
+
+        // Soft roll forward from the inner framework reference [specified] to app's 5.1.1 (defaults)
```

Comment:
Comment should refer to these all being soft rollforward failing cases

Reply:
I changed the comments to be much more informative of what the test is doing.